app/controllers/social_previews_controller.rb

```diff
@@ -38,6 +38,7 @@ def organization
 
   def tag
     @tag = Tag.find(params[:id])
+    @compare_hex = Color::CompareHex.new([@tag.bg_color_hex || "#000000", @tag.text_color_hex || "#ffffff"])
```

Comment:
This appears to be what happens in the `[nil, nil]` case, so seems unlikely to be wildly wrong (it's black text, with black border, on a white background).

Reply:
I wonder if we should move the default values into `Tag` itself, i.e. via custom accessors? Conceptually this makes a bit more sense to me, so tags manage both the values as well as their fallbacks if no values are available. Definitely not a blocker though, we have many places that don't follow a "tell, don't ask" coding style.

Reply:
checking the places we are using the color comparer already - it's usually either via a `user_colors` call, or hard coded decision (this code looks a lot like the same call in the [TagTag](https://github.com/forem/forem/blob/main/app/liquid_tags/tag_tag.rb#L27) liquid tag (that's a mouthful).